hudi-client/hudi-client-common/src/main/java/org/apache/hudi/async/AsyncCompactService.java

```diff
@@ -78,13 +81,26 @@ protected Pair<CompletableFuture, ExecutorService> startService() {
 
         while (!isShutdownRequested()) {
```

Comment:
This portion of the code is identical b/w Clustering/Compaction services. 
Let's abstract it into a shared base class to avoid duplication.